plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/connection/connection.py

```diff
@@ -35,3 +40,56 @@ def clean_up_fqdn(self, fqdn):
     def test(self):
         self.xdr_api.test_connection()
         return {"success": True}
+
+    def test_task(self):
+        self.logger.error("Running a Task Connection Test for Palo Alto Cortex")
+        return_message = "The connection test to Palo Alto Cortex has failed \n"
+
+        # Gets the current time and converts to  Epoch Unix time
+        now = datetime.now(timezone.utc)
+
+        start_time = int((now - timedelta(minutes=5)).timestamp() * 1000)
+        end_time = int(now.timestamp() * 1000)
+        time_sort_field = "creation_time"
+
+        filters = [
+            {"field": time_sort_field, "operator": "gte", "value": start_time},
+            {"field": time_sort_field, "operator": "lte", "value": end_time},
+        ]
+
+        post_body = {
+            "request_data": {
+                "search_from": 0,
+                "search_to": 100,
+                "sort": {"field": time_sort_field, "keyword": "asc"},
+                "filters": filters,
+            }
+        }
+
+        try:
+            _, _, _ = self.xdr_api.get_response_alerts(post_body)
+            message = "The connection test to Palo Alto Cortex was successful"
+            self.logger.info(message)
+            return {"success": True}, message
+
+        except PluginException as error:
+            if error.cause in ("401", "402", "403", "404"):
```

Comment:
I think this error handling will have to change, the get_response_alerts doesn't use the same response handler as the other API call methods.